### PR TITLE
Fake back ends package

### PR DESCRIPTION
**Pull request by mgamis-msft**
# What

Move fake chat back end code from react-composites to a new package called 'fake-backends'

# Why

To make fake chat usable in other modules in the library 

# How Tested

Tested on branch https://github.com/Azure/communication-ui-library/tree/test-chat-bots to create a story with chat participants

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: packages/fake-backends/README.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# fake-backends
+
+This package contains in-memory implementations for serveral Azure Communication Services backends.
+These fake implementations are intended to support several use-cases independent of actual service backends:
+
```
Comment (dmceachernmsft) [inline packages/fake-backends/README.md:5]: Maybe add a note here about storybook?

File: packages/react-composites/package.json [hunk 2]
```diff
@@ -85,6 +85,7 @@
     "@babel/cli": "~7.16.0",
     "@babel/core": "~7.16.0",
     "@babel/preset-env": "7.13.10",
+    "@internal/fake-backends": "1.3.1-beta.1",
```
Comment (dmceachernmsft) [inline packages/react-composites/package.json:88]: is this versioning correct? are we making it all consistent? or should this be 1.0.0?

File: rush.json [hunk 3]
```diff
@@ -112,6 +112,11 @@
       "packageName": "@internal/check-treeshaking",
       "projectFolder": "tools/check-treeshaking",
       "reviewCategory": "tools"
+    },
+    {
+      "packageName": "@internal/fake-backends",
+      "projectFolder": "packages/fake-backends",
+      "reviewCategory": "internal"
```
Comment (dmceachernmsft) [inline rush.json:119]: Should this maybe be `tools`? I am not too sure about what this does.

File: packages/fake-backends/README.md (hunk 1 above)
Comment (mgamis-msft) [reply packages/fake-backends/README.md:5]: The first bullet point is about storybook :)

File: packages/react-composites/package.json (hunk 2 above)
Comment (mgamis-msft) [reply packages/react-composites/package.json:88]: Yes we are making it all consistent.

Comment (prprabhu-ms) [reply packages/react-composites/package.json:88]: Always keep all packlets consistent, for now.

File: packages/react-composites/tests/browser/chat/app/FakeAdapterApp.tsx [hunk 4]
```diff
@@ -24,9 +24,8 @@ import {
   customOnRenderMessage,
   customOnRenderTypingIndicator
 } from './CustomDataModel';
-import { FakeChatClient } from './fake-back-end/FakeChatClient';
-import { Model } from './fake-back-end/Model';
-import { IChatClient } from './fake-back-end/types';
+// TODO: Should import using path '@internal/fake-backends' from tsconfig but it is not working
```
Comment (prprabhu-ms) [inline packages/react-composites/tests/browser/chat/app/FakeAdapterApp.tsx:27]: Please fix this TODO before merging.

**Review by prprabhu-ms — APPROVED** [review]

File: packages/react-composites/tests/browser/chat/app/FakeAdapterApp.tsx (hunk 4 above)
Comment (prprabhu-ms) [reply packages/react-composites/tests/browser/chat/app/FakeAdapterApp.tsx]: I pushed a commit that _should_ fix the error (I assume) you were seeing in the CI.

Feel free to resolve this comment thread if that succeeds.

Comment (mgamis-msft) [reply packages/react-composites/tests/browser/chat/app/FakeAdapterApp.tsx]: Thanks for your commits on this branch but I'm still seeing this error locally when running `rushx build:e2e:chat`
image

Comment (prprabhu-ms) [reply packages/react-composites/tests/browser/chat/app/FakeAdapterApp.tsx]: Oops. I can't type. Let's have CI try that again.

**Review by dmceachernmsft — APPROVED** [review]

**Pull request merged**